explorer/interpreter/resolve_unformed.cpp

```diff
@@ -20,17 +20,33 @@ auto FlowFacts::TakeAction(Nonnull<const AstNode*> node, ActionType action,
                            SourceLocation source_loc, const std::string& name)
     -> ErrorOr<Success> {
   switch (action) {
-    case ActionType::AddInit:
-      AddInitFact(node);
+    case ActionType::AddInit: {
+      AddFact(node, FormedState::MustBeFormed);
       break;
-    case ActionType::AddUninit:
-      AddUninitFact(node);
+    }
+    case ActionType::AddUninit: {
+      AddFact(node, FormedState::Unformed);
       break;
-    case ActionType::Form:
-      FormFact(node);
+    }
+    case ActionType::Form: {
+      // TODO: Use CARBON_CHECK when we are able to handle global variables.
+      if (facts_.count(node) &&
+          facts_[node].formed_state == FormedState::Unformed) {
+        facts_[node].formed_state = FormedState::MayBeFormed;
+      }
```

Comment:
```suggestion
      auto entry = facts_.find(node);
      if (entry != facts_.end() &&
          entry->second.formed_state == FormedState::Unformed) {
        entry->second.formed_state = FormedState::MayBeFormed;
      }
```

Generally I'd suggest avoiding `.count`, especially when an iterator can be used.